lib/oli/groups.ex

```diff
@@ -19,6 +19,27 @@ defmodule Oli.Groups do
   """
   def list_communities, do: Repo.all(Community)
 
+  @doc """
+  Returns the list of communities that meets the criteria passed in the input.
+
+  ## Examples
+
+      iex> search_communities(%{status: :active})
+      [%Community{status: :active}, ...]
+
+      iex> search_communities(%{global_access: false})
+      []
+  """
+  def search_communities(filter) do
+    conditions =
+      Enum.reduce(filter, false, fn {field, value}, conditions ->
```

Comment:
What about making the reduce a private function that's called here - to make it easier to scan/read. Could be re-used also.

Reply:
I like it :)